### ISSUE TITLE
Speedtest Fail - starting yesterday the 21st

### ISSUE THREAD

**Issue by imzenreally** [breadlysm/speedtest-to-influxdb#36]
been working like a charm - and then its over - happily talking to my influx db - but seems to be unhappy with ookla -- their client change or something?

**Comment by chrisweis** [comment]
Same here, can't seem to figure out why.  Tried to change LOG_TYPE environment variable to "debug" to see more detail but nothing.

Considering a switch over to this: https://github.com/alexjustesen/speedtest-tracker

**Comment by imzenreally** [comment]
> Same here, can't seem to figure out why.  Tried to change LOG_TYPE environment variable to "debug" to see more detail but nothing.
> 
> 
> 
> Considering a switch over to this: https://github.com/alexjustesen/speedtest-tracker

Are you running this container in unraid?

I've been seeing some odd ssl errors that (according to a separate thread) seem related to the docker version 

I'm running unraid 6.12.1 and everything is up to date and current etc

**Comment by chrisweis** [comment]
Nope, I'm running it atop an Ubuntu VM on Proxmox

**Comment by imzenreally** [comment]
Damn - rules out that theory

I should spin up a vm on my proxmox box and install this see if it talks

Maybe my aggressive rules on my local network are screwing things up

Did you see any ssl errors in your logs?

TJV

**Comment by chrisweis** [comment]
Which logs are you seeing the SSL errors?  No SSL errors in my docker logs.  I have:

```
{"log":"2023-06-24 01:44:25,819 [INFO] Speedtest CLI data logger to InfluxDB started...\r\n","stream":"stdout","time":"2023-06-24T01:44:25.819485356Z"}
{"log":"2023-06-24 01:44:25,839 [INFO] Ping data written successfully\r\n","stream":"stdout","time":"2023-06-24T01:44:25.83961178Z"}
{"log":"2023-06-24 01:44:25,861 [INFO] Manual server choice : ID = 10162\r\n","stream":"stdout","time":"2023-06-24T01:44:25.861743637Z"}
{"log":"2023-06-24 01:44:25,861 [INFO] Speedtest Failed :\r\n","stream":"stdout","time":"2023-06-24T01:44:25.86180898Z"}
{"log":"2023-06-24 01:45:25,900 [INFO] Ping data written successfully\r\n","stream":"stdout","time":"2023-06-24T01:45:25.901179612Z"}
{"log":"2023-06-24 01:46:25,969 [INFO] Ping data written successfully\r\n","stream":"stdout","time":"2023-06-24T01:46:25.969823677Z"}
```

**Comment by chrisweis** [comment]
Looks like it stopped working for me 2023-06-21 13:30 (US Pacific): 
![image](https://github.com/breadlysm/SpeedFlux/assets/13931731/1763f78f-f0a2-4849-a617-20ddcfc3b26a)

**Comment by imzenreally** [comment]
I’ll check mine tomorrow and post the ssl errors

**Comment by imzenreally** [comment]
```2023-06-23 20:48:32,590 [INFO] Automatic server choice
2023-06-23 20:48:32,590 [INFO] Speedtest Failed :
2023-06-23 20:48:32,591 [DEBUG] b'[2023-06-23 20:48:32.588] [error] Configuration - SSL connect error (UnknownException)\n[2023-06-23 20:48:32.589] [error] Configuration - Cannot retrieve configuration document (0)\n[2023-06-23 20:48:32.589] [error] ConfigurationError - Could not retrieve or read configuration (Configuration)\n[2023-06-23 20:48:32.589] [error] ConfigurationError - Could not retrieve or read configuration (Configuration)\n{"type":"log","timestamp":"2023-06-24T03:48:32Z","message":"Configuration - Could not retrieve or read configuration (ConfigurationError)","level":"error"}\n'
2023-06-23 20:48:32,591 [DEBUG] b''```

my last successful pull was 6/21 0930 Pacific

**Comment by mattdymes** [comment]
I started having the problem on the same date...  Just switched over to https://github.com/alexjustesen/speedtest-tracker in a docker container. Be aware - the new speedtest-tracker is API or Influx 2.0 only. out of the box it seems like traditional username / password authentication to in Influx database is not an option, it is key only. I will work , but will not provide the same functionality of speedflux

**Comment by imzenreally** [comment]
@mattdymes i've avoided using that because i'm resisting upgrading myinfluxdb - i'm still running ... 1.8 i think

**Comment by imzenreally** [comment]
@mattdymes but the graphs the guy has set up as default are nice - i suppose it would replace my grafana dahsboard

**Comment by funnelcloudservices** [comment]
UnRaid guy here.  yep, Mine started failing as well.

**Comment by imzenreally** [comment]
looks like the sppedtest-tracker thats been linked here a few times has plans to support influxdb v1 - i'm probably bailing on speedflux - repo doesn't appear to be maintained and i'm too lazy to troubleshoot when there are other working options

**Comment by chrisweis** [comment]
This thread helped me upgrade my v1 to v2 using a Docker container: https://community.influxdata.com/t/problem-backup-1-8-restore-2-6/28943

Hypothesis I considered but didn't test: Did Ookla Speedtest increase the maximum frequency of tests, and that began causing it to fail?  Would a decrease to only hourly fix the old Speedflux? 🤔

**Comment by clintkev251** [comment]
Nah that's not it. I only test once a day, and mine have been failing consistently too

**Comment by imzenreally** [comment]
> Would a decrease to only hourly fix the old Speedflux? 🤔

i only ran my speedflux once ever 2 or 3 hours - so i don't think its ookla rate limiting

**Comment by ananyosen** [comment]
Looks like a case of horribly updated `speedtest` binary, the image was last published almost 2 years back (https://github.com/breadlysm/SpeedFlux/pkgs/container/speedflux). Manually upgraded to latest speedtest binary from ookla (https://install.speedtest.net/app/cli/ookla-speedtest-1.2.0-linux-x86_64.tgz) after exec'ing into the container, replaced the binary in `/usr/bin/speedtest` and speedtests are working fine now.

This will not persist across container updates (but that does not seem to be an issue at this point)

![image](https://github.com/breadlysm/SpeedFlux/assets/8464592/5dd7df34-8002-45af-ba68-2490028d4e31)

Edit: There's a fork that's regularly updated (https://hub.docker.com/r/dontobi/speedflux.rpi), will test it over the weekend and migrate if it's a drop-in replacement.

**Comment by imzenreally** [comment]
> There's a fork that's regularly updated

slick - i was just about to fork it and take a whack up making the update - let me know if that works as a drop in replacement

edit: fail as a drop in replacement in unraid - its targetted specifically for ARM's

**Comment by faspina** [comment]
> Looks like a case of horribly updated `speedtest` binary, the image was last published almost 2 years back (https://github.com/breadlysm/SpeedFlux/pkgs/container/speedflux). Manually upgraded to latest speedtest binary from ookla (https://install.speedtest.net/app/cli/ookla-speedtest-1.2.0-linux-x86_64.tgz) after exec'ing into the container, replaced the binary in `/usr/bin/speedtest` and speedtests are working fine now.
> 
> This will not persist across container updates (but that does not seem to be an issue at this point)
> 
> ![image](https://user-images.githubusercontent.com/8464592/249238174-5dd7df34-8002-45af-ba68-2490028d4e31.png)
> 
> Edit: There's a fork that's regularly updated (https://hub.docker.com/r/dontobi/speedflux.rpi), will test it over the weekend and migrate if it's a drop-in replacement.

This worked for me . Downloaded the file, unpack put in it a director and added that directory as a resource to the docker container, then went into the container consol and copied speedtest to the /usr/bin to replace existing one, restarted container it started working

**Comment by funnelcloudservices** [comment]
Oddly enough mine started working again overnight.

**Comment by imzenreally** [comment]
I just reinstalled and tested and its working again

**Comment by clintkev251** [comment]
Yup, mine has also just randomly started working again. No changes, no updates.

**Comment by sethwv** [comment]
> Looks like a case of horribly updated `speedtest` binary, the image was last published almost 2 years back (https://github.com/breadlysm/SpeedFlux/pkgs/container/speedflux). Manually upgraded to latest speedtest binary from ookla (https://install.speedtest.net/app/cli/ookla-speedtest-1.2.0-linux-x86_64.tgz) after exec'ing into the container, replaced the binary in `/usr/bin/speedtest` and speedtests are working fine now.
> 
> This will not persist across container updates (but that does not seem to be an issue at this point)
> 
> ![image](https://user-images.githubusercontent.com/8464592/249238174-5dd7df34-8002-45af-ba68-2490028d4e31.png)
> 
> Edit: There's a fork that's regularly updated (https://hub.docker.com/r/dontobi/speedflux.rpi), will test it over the weekend and migrate if it's a drop-in replacement.

I've made a PR to the effect of changing the binary even though it's unlikely to be merged as this project is unfortunately so inactive - if you need a drop-in replacement for unraid then for now you're welcome to build an image of your own with my PR fork or use my repo at https://hub.docker.com/repository/docker/sethwv/speedflux/
